agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -455,26 +466,42 @@ func (cs *CredentialSpecResource) writeS3File(writeFunc func(file oswrapper.File
 	if err != nil {
 		return err
 	}
-	defer temp.Close()
 
 	err = writeFunc(temp)
 	if err != nil {
 		return err
 	}
 
-	err = temp.Chmod(os.FileMode(filePerm))
+	// Persist the file to disk.
+	err = temp.Sync()
 	if err != nil {
 		return err
 	}
 
-	// Persist the file to disk.
-	err = temp.Sync()
+	input, err := ioutil.ReadFile(temp.Name())
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filePath, input, 0644)
+	if err != nil {
+		return err
+	}
+
+	err = acl.Chmod(filePath, filePerm);
```

Comment:
why won't the built in `os.Chmod()` work for our use case?

Reply:
os.chmod() has issues on windows . That's the reason i used an external library built for windows and replaced the existing os.chmod call . But i just realized that we don't need to change the permissions explicitly. I tested it and it works. Not sure why the previous developer had this call but i feel its not needed. I removed this.